### PR TITLE
TWave fix

### LINKED ISSUE #8887: Subtracting waves in optics module
The Optics module enables working with waves.

However, subtracting the waves which follows intuitively results in the result being an instance of the `sympy.core.add.Add` class. This stops the user from accessing properties of the resultant wave like amplitude, phase etc. 

This can be fixed by simply returning the result of the  `__add__(self, other)` method with the amplitude of `other` multiplied by `-1`.

Is it a good idea to make this change ?

``` python
In [1]: from sympy.physics.optics import *
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-1-1d52e0c05aa5> in <module>
----> 1 from sympy.physics.optics import *

AttributeError: module 'sympy.physics.optics' has no attribute 'polarizing_beam_splitte'

In [2]: from sympy.physics.optics import TWave

In [3]: A,f,phi = symbols('A,f,phi')

In [4]: w1 = TWave(A, f)

In [5]: w2 = TWave(A, f, phi)

In [6]: w1+w2
Out[6]: TWave(sqrt(2*A**2*cos(phi) + 2*A**2), f, atan2(A*cos(phi) + A, A*sin(phi)))

In [7]: _.amplitude
Out[7]: 
   ____________________
  ╱    2             2 
╲╱  2⋅A ⋅cos(φ) + 2⋅A  

In [8]: w1 - w2
Out[8]: TWave(A, f) - TWave(A, f, phi)

In [9]: type(_)
Out[9]: sympy.core.add.Add

In [10]: (w1 - w2).amplitude
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-10-cea1328571ac> in <module>
----> 1 (w1 - w2).amplitude

AttributeError: 'Add' object has no attribute 'amplitude'
```

### PR DESCRIPTION
**Pull request by kevinventullo**
Fixes #8887.

Implements negation, subtraction, and scalar multiplication of TWaves. It looks like @darshanime has already done some work, and I incorporated some fixes and testing he made. 

I also noticed a bug in TWave addition: the sin's and cos's were switched around in the arguments to atan2. You can check this by adding a zero amplitude wave to some wave and using the identity 
`atan2(sin(x), cos(x)) = x (mod 2*pi)`

### CODE REVIEW

**Comment by kevinventullo** [comment]
@debugger22 Hi, I am done making changes and the checks have passed. Is this okay to merge?

File: sympy/physics/optics/waves.py [hunk 1]
```diff
@@ -10,7 +10,7 @@
 
 __all__ = ['TWave']
 
-from sympy import (sympify, pi, sin, cos, sqrt, Symbol, S,
+from sympy import (sympify, pi, sin, cos, sqrt, Number, Symbol, S,
     symbols, Derivative, atan2)
 from sympy.core.expr import Expr
 from sympy.physics.units import c
```
Comment (darshanime) [inline sympy/physics/optics/waves.py]: Kindly change `:math:\omega is angular velocity` to `:math:\omega is angular frequency` on line 27.

**Comment by darshanime** [comment]
Don't forget to clean the commit history.

**Comment by kevinventullo** [comment]
So for some reason I'm failing tests in the combinatorics and polys modules, but I never altered anything in these modules.

**Comment by debugger22** [comment]
> So for some reason I'm failing tests in the combinatorics and polys modules, but I never altered anything in these modules.

That's an irrelevant failure. I restarted the test.

**Comment by czgdp1807** [comment]
This looks like a good addition.
@kevinventullo Would you still like to work on this?

**Pull request closed without merging**